src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -138,34 +150,41 @@ def evaluate(
         operational_settings_used = TimeSeriesInt(
             timesteps=variables_map.time_vector, values=[0] * len(variables_map.time_vector), unit=Unit.NONE
         )
-        operational_settings_results: Dict[int, Dict[str, EcalcModelResult]] = defaultdict(
+
+        operational_settings_results: Dict[datetime, Dict[int, Dict[str, EcalcModelResult]]] = defaultdict(
             dict
         )  # map operational settings index and consumer id to consumer result.
 
-        for period, operational_settings in temporal_operational_settings.items():
-            variables_map_for_period = variables_map.get_subset_from_period(period)
-            start_index, end_index = period.get_timestep_indices(variables_map.time_vector)
+        # for period, operational_settings in temporal_operational_settings.items():
+        for timestep_index, timestep in enumerate(variables_map.time_vector):
```

Comment:
I guess this is fine? Is this something we want to do for all v2 models? i.e. in compressor and pump also? So all models are evaluated with only a single timestep?

Reply:
Waoh. I didnt expect you to agree on this, since it is, as you say, changing in general the way we evaluate stuff to timesteps instead of periods.

I didnt add it to the PR, thought about discussing it, but in general, it was (imo) so complicated to get things right with temporal models overlapping temporal models and so on, that at least the initial approach had to be timestep by timestep, to be sure to get it right...

However, I realized after mentioning it the other day, that this will probably be in many case hard to cache since the runtime variables change for each timestep, so it will most definitely be slower than the previous solution. Therefore, it makes more sense to be to bring this "timestep by timestep" even one more layer/level out, to create eCalc models for each timestep (WITHOUT temporal models to deal with, only do that at evaluation/preprocessing step)...but there will be a performance penalty I think. We should definitely discuss it...

I havent thought too much about it wrt. consequences of performance penalty etc, but it will remove a lot of unnecessary complexity (for developers) and potentially add benefits for users, that I think might be worth it...